x-pack/legacy/plugins/maps/public/layers/styles/vector/vector_style.js

```diff
@@ -218,43 +218,56 @@ export class VectorStyle extends AbstractStyle {
 
   async pluckStyleMetaFromSourceDataRequest(sourceDataRequest) {
     const features = _.get(sourceDataRequest.getData(), 'features', []);
-    if (features.length === 0) {
-      return {};
-    }
-
-    const dynamicProperties = this.getDynamicPropertiesArray();
 
     const supportedFeatures = await this._source.getSupportedShapeTypes();
-    const isSingleFeatureType = supportedFeatures.length === 1;
-    if (dynamicProperties.length === 0 && isSingleFeatureType) {
-      // no meta data to pull from source data request.
-      return {};
-    }
-
-    let hasPoints = false;
-    let hasLines = false;
-    let hasPolygons = false;
-    for (let i = 0; i < features.length; i++) {
-      const feature = features[i];
-      if (!hasPoints && POINTS.includes(feature.geometry.type)) {
-        hasPoints = true;
-      }
-      if (!hasLines && LINES.includes(feature.geometry.type)) {
-        hasLines = true;
-      }
-      if (!hasPolygons && POLYGONS.includes(feature.geometry.type)) {
-        hasPolygons = true;
+    let hasFeatureType;
+    if (supportedFeatures.length > 1) {
```

Comment:
Yes this is necessary. If `supportedFeatures` is only a single type then there is no need to loop through `features` because we already know the type.